chia/timelord/timelord.py

```diff
@@ -953,100 +953,94 @@ async def _do_process_communication(
                         self.vdf_failures_count += 1
                     break
 
-                msg = ""
-                try:
-                    msg = data.decode()
-                except Exception:
-                    pass
-                if msg == "STOP":
```

Comment:
the only thing we use `msg` for is to compare it to "STOP". Currently we interpret these 4 bytes as utf-8, just to convert it to a string we then compare against "STOP". The vast majority of times, this conversion fails, because the bytes aren't utf-8, that's why we just ignore any exception coming out of the `decode()` function.

There's no real point of converting the bytes to a string, instead, we can just compare it to the *bytes* b"STOP".